extensions/wikia/CoppaTool/RenameIPProcess.class.php

```diff
@@ -0,0 +1,1094 @@
+						$blockTokens = explode( '=', $tokens[2], 2 );
+
+						if (
+							count( $blockTokens ) == 2 &&
+							$blockTokens[0] === self::PHALANX_BLOCK_TAG &&
+							is_numeric( $blockTokens[1] )
+						) {
+							$this->mPhalanxBlockId = (int)$blockTokens[1];
+						}
+					}
+			}
+
+			if ( $repeating ) {
+				$this->addWarning( wfMessage( 'userrenametool-warn-repeat', $this->mRequestData->oldUsername, $this->mRequestData->newUsername )->inContentLanguage()->text() );
+				// Swap the uids because the real user ID is the new user ID in this special case
+				$fakeUid = $uid;
+				$uid = User::idFromName( $newTitle->getText() );
+			} else {
+				// In the case other than repeating the process drop an error
+				$this->addError( wfMessage( 'userrenametool-errorexists', $newUser->getName() )->inContentLanguage()->text() );
+				return false;
+			}
+		}
+
+		// Execute Warning hook (arguments the same as in the original Renameuser extension)
+		if ( !$this->mActionConfirmed ) {
+			Hooks::run( 'UserRename::Warning', array( $this->mRequestData->oldUsername, $this->mRequestData->newUsername, &$this->mWarnings ) );
+		}
+
+		$this->mOldUsername = $oldUser->getName();
+		$this->mNewUsername = $newUser->getName();
+		$this->mUserId = (int)$uid;
+		$this->mFakeUserId = $fakeUid;
+
+		$this->addInternalLog( "setup: uid={$this->mUserId} fakeuid={$this->mFakeUserId} old={$this->mOldUsername} new={$this->mNewUsername}" );
+
+		// If there are only warnings and user confirmed that, do not show them again
+		// on success page ;-)
+		if ( $this->mActionConfirmed ) {
+			$this->mWarnings = array();
+		} elseif ( count( $this->mWarnings ) ) {
+			// in case action is not confirmed and there are warnings display them and wait for confirmation before running the process
+			return false;
+		}
+
+		return empty( $this->mErrors );
+	}
+
+	/**
+	 * Runs the whole rename process, schedules background jobs/tasks if needed.
+	 *
+	 * @return bool True if the process succeded
+	 */
+	public function run() {
+		// Make sure the process will not be stopped in the middle
+		set_time_limit( 3600 ); // 1h
+		ignore_user_abort( true );
+		ini_set( "max_execution_time", 3600 ); // 1h
+
+		if ( !$this->setup() ) {
+			return false;
+		}
+
+		// Execute the worker
+		$status = false;
+
+		try {
+			$status = $this->doRun();
+		} catch ( Exception $e ) {
+			$this->addLog( $e->getMessage() . ' in ' . $e->getFile() . ' at line ' . $e->getLine() );
+			$this->addError( wfMessage( 'userrenametool-error-cannot-rename-unexpected' )->inContentLanguage()->text() );
+		}
+
+		// Analyze status
+		if ( !$status ) {
+			$this->addMainLog(
+				'fail',
+				RenameUserLogFormatter::fail(
+					$this->mRequestorName,
+					$this->mOldUsername,
+					$this->mNewUsername,
+					$this->mReason
+				)
+			);
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Do the whole dirty job of renaming user
+	 *
+	 * @return bool True if the process succeded
+	 */
+	private function doRun() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

[project-id]: wikia:app